ddtrace/llmobs/_integrations/pydantic_ai.py

```diff
@@ -145,6 +129,58 @@ def _llmobs_set_tags_tool(
         if not span.error:
             span._set_ctx_item(OUTPUT_VALUE, getattr(response, "content", ""))
 
+    def _tag_agent_manifest(self, span: Span, kwargs: Dict[str, Any], agent: Any) -> None:
+        if not agent:
+            return
+
+        manifest = {}
+        manifest["framework"] = "PydanticAI"
+        if hasattr(agent, "name"):
+            manifest["name"] = agent.name or "PydanticAI Agent"
+        if span.get_tag("pydantic_ai.request.model"):
+            manifest["model"] = span.get_tag("pydantic_ai.request.model")
+        if span.get_tag("pydantic_ai.request.provider"):
+            manifest["model_provider"] = span.get_tag("pydantic_ai.request.provider")
+        if hasattr(agent, "model_settings"):
+            manifest["model_settings"] = agent.model_settings
+        if hasattr(agent, "_instructions"):
+            manifest["instructions"] = agent._instructions
+        if hasattr(agent, "_system_prompts"):
+            manifest["system_prompts"] = agent._system_prompts
+        if hasattr(agent, "_function_tools"):
+            manifest["tools"] = self._get_agent_tools(agent._function_tools)
+        if kwargs.get("deps", None):
+            agent_dependencies = kwargs.get("deps", None)
+            manifest["dependencies"] = (
+                agent_dependencies.__dict__ if hasattr(agent_dependencies, "__dict__") else agent_dependencies
+            )
+
+        span._set_ctx_item(AGENT_MANIFEST, manifest)
+
+    def _get_agent_tools(self, tools: Any) -> List[Dict[str, Any]]:
+        if not tools:
+            return []
+        formatted_tools = []
+        for tool_name, tool_instance in tools.items():
+            tool_dict = {}
+            tool_dict["name"] = tool_name
+            if hasattr(tool_instance, "description"):
+                tool_dict["description"] = tool_instance.description
+            function_schema = getattr(tool_instance, "function_schema", None)
+            json_schema = getattr(function_schema, "json_schema", None)
+            required_params = {param: True for param in json_schema.get("required", [])}
```

Comment:
```suggestion
            json_schema = getattr(function_schema, "json_schema", {})
            required_params = {param: True for param in json_schema.get("required", [])}
```
i think we should default to an empty dict if we wanna try and access stuff from it